massa-network/src/peer_info_database.rs

```diff
@@ -349,114 +382,60 @@ impl PeerInfoDatabase {
                         return false;
                     }
-                    if let Some(last_failure) = p.last_failure {
-                        if let Some(last_alive) = p.last_alive {
-                            if last_alive > last_failure {
-                                return true;
-                            }
-                        }
-                        return now
-                            .saturating_sub(last_failure)
-                            .saturating_sub(self.wakeup_interval)
-                            > MassaTime::from(0u64);
-                    }
-                    true
+                    peer_ready(self.wakeup_interval, now, p)
                 })
                 .copied()
                 .collect();
             sorted_peers
                 .sort_unstable_by_key(|&p| (p.last_failure, std::cmp::Reverse(p.last_alive)));
-            res_ips.extend(
-                sorted_peers
-                    .into_iter()
-                    .take(available_slots_bootstrap)
-                    .map(|p| p.ip),
-            )
-        }
 
-        if available_slots_nonbootstrap > 0 {
-            let now = MassaTime::compensated_now(self.clock_compensation)?;
-            let mut sorted_peers: Vec<PeerInfo> = self
-                .peers
-                .values()
-                .filter(|&p| {
-                    if !(!p.bootstrap && p.advertised && !p.banned && !p.is_active()) {
-                        return false;
-                    }
-                    if let Some(last_failure) = p.last_failure {
-                        if let Some(last_alive) = p.last_alive {
-                            if last_alive > last_failure {
-                                return true;
-                            }
-                        }
-                        return now
-                            .saturating_sub(last_failure)
-                            .saturating_sub(self.wakeup_interval)
-                            > MassaTime::from(0u64);
-                    }
-                    true
-                })
-                .copied()
-                .collect();
-            sorted_peers
-                .sort_unstable_by_key(|&p| (p.last_failure, std::cmp::Reverse(p.last_alive)));
-            res_ips.extend(
-                sorted_peers
-                    .into_iter()
-                    .take(available_slots_nonbootstrap)
-                    .map(|p| p.ip),
-            )
+            Ok(sorted_peers.into_iter().take(avaible_slots).map(ip))
+        } else {
+            Ok(Vec::new().into_iter().take(avaible_slots).map(ip))
         }
-
-        Ok(res_ips)
+    }
+    /// Sorts peers by ( last_failure, rev(last_success) )
+    /// and returns as many peers as there are available slots to attempt outgoing connections to.
+    pub fn get_out_connection_candidate_ips(&self) -> Result<Vec<IpAddr>, NetworkError> {
+        // TODO why is this comment here ??
```

Comment:
I wasn't really able to find when it was written :/

Reply:
Remove it, it looks like an old fossil out of time